mitodl/mit-learn
# Professional Education ETL Pipeline

### What are the relevant tickets?
Closes https://github.com/mitodl/hq/issues/4731

Related `ol-infrastructure` PR: https://github.com/mitodl/ol-infrastructure/pull/2536

### Description (What does it do?)
- Adds a new ETL pipeline to ingest courses and programs directly from Professional Education's own API feed instead of Prolearn's.
- Excludes PE courses from being ingested by the Prolearn ETL pipeline.


### Screenshots (if appropriate):
![Screenshot 2024-07-08 at 8 59 20 AM](https://github.com/mitodl/mit-open/assets/187676/859d848f-e902-47d6-adf9-f3ddda2e37e2)


### How can this be tested?
- Set the following values in your backend .env file:
  ```
   PROLEARN_CATALOG_API_URL=https://prolearn.mit.edu/graphql
   MITPE_BASE_API_URL=https://professional.mit.edu/
  ```
-  Run `./manage.py backpopulate_prolearn_data`
-  When it is done, there should be roughly >130 learning resources with `etl_source=prolearn` and `offered_by=mitpe`.  They will all be courses and not programs.  http://api.open.odl.local:8063/admin/learning_resources/learningresource/?o=2&offered_by__code__exact=mitpe
- If you don't already have a populated featured learning path for Professional Education, run `./manage.py populate_featured_lists` to create one

- Switch to this branch.  
- Set `MITPE_API_ENABLED=True` in your backend .env file, restart containers.
- Run `./manage.py backpopulate_mitpe_data` followed by `./manage.py backpopulate_prolearn_data`.  When it is done, most/all the same courses returned above should now have `etl_source=mitpe` and have new `readable_id` values.   Some that were courses in prolearn should be unpublished, and created as new programs instead of courses.  There should not be any published dupes from different sources (prolearn and mitpe).  Some will be unpublished now due to changed criteria (past enrollment end date or past end date).
- If you go to `http://open.odl.local:8062/learningpaths` and open the Professional Education featured learning path, it will be empty.  Run the following to reassign the old featured courses from prolearn to their new equivalents
```
./manage.py transfer_list_resources course course title prolearn mitpe
./manage.py transfer_list_resources course program title prolearn mitpe
```
- Run `./manage.py update_index --all`
- Once complete, you should have courses/programs in the MITPE featured learning path again.  These commands will need to be run on RC/production as well to retain the current featured lists.
